src/utils/user/notification.ts

```diff
@@ -1,5 +1,30 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { FirebaseMessagingTypes } from '@react-native-firebase/messaging';
+import moment from 'moment';
 import { Vibration } from 'react-native';
 import { client } from '../../config/axiosConfig';
+import { NotificationType } from '../../constants/NotificationType';
+import { EmergencyInfo } from '../../screens/EmergencyInfoScreen';
+
+export type NotificationFeed = {
+  emergency: Notification[];
+  reminder: Notification[];
+};
+
+export type Notification =
+  | {
+      message: string;
+      messageType: NotificationType;
+    }
+  | EmergencyNoti;
+
+export type EmergencyNoti = EmergencyInfo & { messageType: NotificationType };
+
+export const determineMessageType = (data) => {
+  if (data.latitude) return NotificationType.EMERGENCY;
+  if (data.isCancelled) return NotificationType.EMERGENCY_CANCEL;
```

Comment:
Given that notification payload can only be sent in string format, i'm not sure if `data.isCancelled` will be boolean

Reply:
This property only exists in cancel notification, right? If so, it will be undefined for other noti. And that means it will be false. @sad7898